pkgs/_analyzer_macros/bin/server.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:_analyzer_macros/macro_implementation.dart';
+import 'package:analysis_server/starter.dart';
+import 'package:analyzer/src/summary2/macro_injected_impl.dart' as injected;
+import 'package:macro_service/macro_service.dart';
+
+/// Analysis server with `dart_model` implementation injected.
+///
+/// Run with your IDE by compiling and placing in the SDK your IDE is using,
+/// for example:
+///
+/// dart compile kernel bin/server.dart
+/// cp bin/server.dill ~/opt/dart-sdk-be/bin/snapshots/analysis_server.dart.snapshot
+///
+/// Then restart, VSCode: Ctrl+Shift+P, Restart Analysis Server.
+///
+/// Only works for one path, see `packageConfig` path hardcoded below.
+void main(List<String> args) async {
+  final home = Platform.environment['HOME']!;
+  injected.macroImplementation = await AnalyzerMacroImplementation.start(
```

Comment:
I am curious how we are going to handle enabling/disabling this in general?

If we are going to attempt to support old and new macros together, then the way we are injecting things right now won't really work, because it is a global setting, and you might have multiple projects open with different macro implementations.

Reply:
I'm sure I wrote a reply to this, not sure where it went...

My expectation is that we'll switch from one to the other without supporting both, and possibly with some downtime in between when we support neither.

It would be more work+complexity to support both for external users, e.g. we'd have to do something about `package:macros` being used in two different ways, or use two different packages.

For our own use for development I guess one or the other is enough, or we can add something more fine-grained if needed.

Reply:
An atomic switch will probably be a bit tricky given all the tests which work with the existing API as well as the JSON macro (which lives in the SDK but we could move it out which might help).

It might still be the easiest answer though.